src/test/java/com/google/api/generator/gapic/composer/ResourceNameHelperClassComposerTest.java

```diff
@@ -227,5 +227,62 @@ public void generateResourceNameClass() {
           + "  public Builder toBuilder() {\n"
           + "    return new Builder(this);\n"
           + "  }\n"
+          + "\n"
+          + "  public static FoobarName of(String project, String foobar) {\n"
+          + "    return newProjectFoobarBuilder().setProject(project).setFoobar(foobar).build();\n"
+          + "  }\n"
```

Comment:
this `newProjectFoobarBuilder` seems not defined above, there is a `ProjectFoobarBuilder` instead, should it aslo start with `new` name? 

Also `ProjectFoobarBuilder` and `newBuilder` both return `new Builder()` is this intended? Thanks.

Reply:
Fixed `newProjectFoobarBuilder`. Yes, returning `new Builder()` for both is intended, e.g. in [this class](https://github.com/googleapis/java-dataproc/blob/master/proto-google-cloud-dataproc-v1/src/main/java/com/google/cloud/dataproc/v1/AutoscalingPolicyName.java).